### PR TITLE
pytorch_transformers -> transformers

### PR DESCRIPTION
**Pull request by maksym-del**
Fixes https://github.com/allenai/allennlp/issues/3448.

Breaking changes that were introduced in the new "transofrmers" release should not affect existing allennlp codebase.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -53,6 +53,7 @@ word2number>=1.1
 # To use the BERT model
 pytorch-pretrained-bert>=0.6.0
 pytorch-transformers==1.1.0
```
Comment (matt-gardner) [inline requirements.txt]: Can't you remove this one?

File: setup.py [hunk 2]
```diff
@@ -132,6 +132,7 @@
         "word2number>=1.1",
         "pytorch-pretrained-bert>=0.6.0",
         "pytorch-transformers==1.1.0",
```
Comment (matt-gardner) [inline setup.py]: And this.

**Comment by nicola-decao** [comment]
This is another place:
https://github.com/allenai/allennlp/blob/5238561412d1f63fa47512b7d6726dee9ed5ae88/allennlp/modules/token_embedders/bert_token_embedder.py#L16
However, this needs a more careful edit since if you just change that it brakes. Indeed, `BertModel.forward` works differently in the new version.

**Comment by maksym-del** [comment]
Yes, ``pytorch_pretrained_bert`` -> ``transformers`` switch is harder then ``pytorch_transformers``->``transformers``. Existing bert_srl model depends on it. 

Let this PR be just ``pytorch_transformers``->``transformers`` migration and upgrading from ``pytorch_pretrained_bert`` can have a separate PR.

**Comment by matt-gardner** [comment]
I agree, I was just commenting on removing the `pytorch-transformers` dependency, not the `pretrained-bert` one.

**Comment by Evpok** [comment]
Regarding this, is the hard-pin to 2.1.1 really necessary?

**Comment by maksym-del** [comment]
@Evpok unless we want to constantly monitor upcoming releases for breaking changes, I believe so.

**Comment by Evpok** [comment]
Well, I respect your choices but surely at some point you will have to monitor this anyway, won't you? So why not just specify a minimum version and only restrict the maximum version if/when tests break?

In any case I am mostly asking because due to python packaging idiosyncrasies, it is basically impossible to override allennlp requirements, so this kind of hard pin is really inconvenient for downstream consumers of both allennlp and transformers. I completely understand if this is inconvenient for you, though.

**Comment by maksym-del** [comment]
@Evpok I see your point regarding specifying a range of versions.

For the transformers library however, there are basically 2 versions: v2.0.0 and v2.1.1. https://github.com/allenai/allennlp/pull/3453 accounts for 2.1.1 and breaks in case of v2.0.0 because of breaking changes. I believe that users should just upgrade to the new version and update their code accordingly (wrt changes in transformers).

Regarding the dependencies issue, running ``pip install transformers=2.1.1`` from allennlp's conda env successfully uninstalls the old version and installs the specified one in my case! You should open a new issue if it is not the case for your setup for some reason.

**Comment by matt-gardner** [comment]
Actually, I think @Evpok is right - we try to be as loose as possible in our bounds, for the reasons that @Evpok brings up.  I thought about this during my review, but ignored it because the previous `pytorch-transformers` dependency was pinned.  But both of them should probably have been `>=`, not `==`.  We get notified very quickly when an updated dependency breaks our build, and it's easy to catch it then and pin it.  @maksym-del, can you change this in #3453?

**Comment by maksym-del** [comment]
@Evpok @matt-gardner sure, I made a commit.

**Comment by Evpok** [comment]
Thanks @matt-gardner, sorry @maksym-del :-( and it was working fine in my setup (which uses virtualenv instead of conda but it doesn't matter), I was just anticipating the future releases as I have been burned by that kind of things in the past. In any case, thank you both for your great work on this library :-)

**Comment by julien-c** [comment]
By the way, if we break your build when releasing a new minor, we also want to know it as fast as possible 🤗 

So feel free to open an issue or send an email if that's the case!

**Pull request merged**